### PR TITLE
Tsify generated asm docs

### PR DESCRIPTION
**Pull request by jeremy-rifkin**

### CODE REVIEW

File: etc/scripts/docenizers/docenizer-6502.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 import urllib.request
 
 
-DOC_URL_BASE = "https://raw.githubusercontent.com/mist64/c64ref/master/6502/"
+DOC_URL_BASE = "https://raw.githubusercontent.com/mist64/c64ref/4274bd8782c5d3b18c68e6b9479b0ec751eb96b1/Source/6502/"
```
Comment (mattgodbolt) [inline etc/scripts/docenizers/docenizer-6502.py:10]: nice!

File: etc/scripts/docenizers/docenizer-amd64.py [hunk 2]
```diff
@@ -19,8 +19,8 @@
 parser.add_argument('-i', '--inputfolder', type=str,
                     help='Folder where the input files reside as .html. Default is ./asm-docs/',
                     default='asm-docs')
-parser.add_argument('-o', '--outputpath', type=str, help='Final path of the .js file. Default is ./asm-docs.js',
-                    default='./asm-docs.js')
+parser.add_argument('-o', '--outputpath', type=str, help='Final path of the .ts file. Default is ./asm-docs.ts',
```
Comment (mattgodbolt) [inline etc/scripts/docenizers/docenizer-amd64.py:22]: can you update the comment too?

File: etc/scripts/docenizers/docenizer-arm32.py [hunk 3]
```diff
@@ -19,8 +19,8 @@
 parser.add_argument('-i', '--inputfolder', type=str,
                     help='Folder where the input files reside as .xml. Default is ./asm-docs-arm/',
                     default='asm-docs-arm')
-parser.add_argument('-o', '--outputpath', type=str, help='Final path of the .js file. Default is ./asm-docs-arm.js',
-                    default='./asm-docs-arm.js')
+parser.add_argument('-o', '--outputpath', type=str, help='Final path of the .ts file. Default is ./asm-docs-arm.ts',
```
Comment (mattgodbolt) [inline etc/scripts/docenizers/docenizer-arm32.py:22]: do you mind updating the comment while you're here please?

File: lib/asm-docs/generated/asm-docs-6502.ts [hunk 4]
```diff
@@ -1,233 +1,235 @@
-export function getAsmOpcode(opcode) {
+import {AssemblyInstructionInfo} from '../base';
+
+export function getAsmOpcode(opcode: string | undefined): AssemblyInstructionInfo | undefined {
     if (!opcode) return;
     switch (opcode.toUpperCase()) {
         case "ADC":
             return {
-                "html": "<p>This instruction adds the value of memory and carry from the previous operation to the value of the accumulator and stores the result in the accumulator.</p><p>This instruction affects the accumulator; sets the carry flag when the sum of a binary add exceeds 255 or when the sum of a decimal add exceeds 99, otherwise carry is reset. The overflow flag is set when the sign or bit 7 is changed due to the result exceeding +127 or -128, otherwise overflow is reset. The negative flag is set if the accumulator result contains bit 7 on, otherwise the negative flag is reset. The zero flag is set if the accumulator result is 0, otherwise the zero flag is reset.</p>",
+                "html": "<p>This instruction adds the value of memory and carry from the previous operation to the value of the accumulator and stores the result in the accumulator.</p>",
```
Comment (mattgodbolt) [inline lib/asm-docs/generated/asm-docs-6502.ts:8]: Do we know why the second paragraph got dropped here in all cases? It's probably better but I wanted to make sure it's intentional.

**Review by mattgodbolt — APPROVED** [review]
Super minor comments only ! thanks mate

File: lib/asm-docs/generated/asm-docs-6502.ts (hunk 4 above)
Comment (jeremy-rifkin) [reply lib/asm-docs/generated/asm-docs-6502.ts:8]: Ah thank you, I was going to revisit this. It looks like the second paragraph is present in the reference but maybe the format changed so I'll update the docenizer :)

File: etc/scripts/docenizers/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "../../../tsconfig.json",
+    "compilerOptions": {
+      "lib": ["es2021"],
```
Comment (jeremy-rifkin) [inline etc/scripts/docenizers/tsconfig.json:4]: This is needed for `String.prototype.replaceAll`. Would be worth looking into for the backend tsconfig as well (when I tried it caused `performance` to not be recognized though)

**Review by RubenRBS — APPROVED** [review]

**Comment by dkm** [comment]
this is live :rocket:

**Pull request merged**